internal/venice-client-common/src/main/java/com/linkedin/venice/compression/ZstdWithDictCompressor.java

```diff
@@ -21,30 +22,23 @@
 
 
 public class ZstdWithDictCompressor extends VeniceCompressor {
-  private ZstdCompressCtx compressor;
-  private byte[] dictionary;
+  private ZstdDictCompress compressorDictionary;
+  private ZstdDictDecompress decompressorDictionary;
 
   public ZstdWithDictCompressor(final byte[] dictionary, int level) {
     super(CompressionStrategy.ZSTD_WITH_DICT);
-    this.dictionary = dictionary;
-    compressor = new ZstdCompressCtx().loadDict(this.dictionary).setLevel(level);
+    this.compressorDictionary = new ZstdDictCompress(dictionary, level);
+    this.decompressorDictionary = new ZstdDictDecompress(dictionary);
   }
 
   @Override
   public byte[] compress(byte[] data) {
-    return compressor.compress(data);
+    return Zstd.compress(data, compressorDictionary);
   }
 
   @Override
   public ByteBuffer compress(ByteBuffer data) {
-    if (data.isDirect()) {
-      // TODO: It might be a decent refactor to add a pool of direct memory buffers so as to always leverage the this
-      // interface and copy the results of the compression back into the passed in ByteBuffer. That would avoid
-      // some of the data copy going on here.
-      return compressor.compress(data);
-    } else {
-      return ByteBuffer.wrap(compressor.compress(ByteUtils.extractByteArray(data)));
-    }
+    return Zstd.compress(data, compressorDictionary);
```

Comment:
This is interesting.  Does it work on direct memory here?

Reply:
I was surprised too - but in the reverse direction. If I traverse the call stack, I reach this function that says `srcBuf` must be direct
```java
    /**
     * Compresses the data in buffer 'srcBuf'
     *
     * @param srcBuf the source buffer - must be direct. It is assumed that the `position()` of the
     *               buffer marks the beginning of the uncompressed data to be compressed, and that
     *               the `limit()` of this buffer marks its end.
     *               <p>
     *               When this method returns successfully, its `position()` will be set to its initial `limit()`.
     *               </p>
     * @return A newly allocated direct ByteBuffer containing the compressed data.
     */
    public ByteBuffer compress(ByteBuffer srcBuf) throws ZstdException {
```
But in the test case, we use non-direct buffer (`ByteBuffer.wrap`)

I modified a test case to test direct BB, non-direct BB and byte[] cases and this is getting enforced in the Zstd lib. Will fix and update it